### DISCUSSION TITLE
Disable JPH_FLOATING_POINT_EXCEPTIONS_ENABLED

### DISCUSSION THREAD

**Discussion by SSNTails** [jrouwe/JoltPhysics#997] [Q&A]
How do I properly disable JPH_FLOATING_POINT_EXCEPTIONS_ENABLED ? I edited CMakeLists.txt to read as such:
`
option(FLOATING_POINT_EXCEPTIONS_ENABLED "Enable floating point exceptions" OFF)
`

But when I build my application and link jolt.lib, I get a runtime error that there is a mismatch with JPH_FLOATING_POINT_EXCEPTIONS_ENABLED. Of course, I can add this preprocessor define to my own application build pipeline and it works, but I want to disable this feature. (It's causing exceptions for stuff like comparing dot products of near-identical vectors)

**Comment by SSNTails** [comment]
I found that I can fix it by completely removing these lines completely from Jolt.cmake, but this is obviously not ideal:

`
# Setting floating point exceptions
if (FLOATING_POINT_EXCEPTIONS_ENABLED AND "${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
	target_compile_definitions(Jolt PUBLIC "$<$<CONFIG:Debug>:JPH_FLOATING_POINT_EXCEPTIONS_ENABLED>")
	target_compile_definitions(Jolt PUBLIC "$<$<CONFIG:Release>:JPH_FLOATING_POINT_EXCEPTIONS_ENABLED>")
endif()
`

**Comment by jrouwe** [comment]
You should just add `-DFLOATING_POINT_EXCEPTIONS_ENABLED=NO` to the cmake commandline when generating the build files for Jolt (or if you're using FetchContent by setting it in your master CMakeLists.txt)
